### PR TITLE
add e2e tests for PYTHON_THREADPOOL_THREAD_COUNT and FUNCTIONS_WORKER_PROCESS_COUNT

### PR DESCRIPTION
**Pull request by pdthummar**
## Description
Adding  e2e tests for PYTHON_THREADPOOL_THREAD_COUNT and FUNCTIONS_WORKER_PROCESS_COUNT.

Fixes #

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/endtoend/test_worker_proccess_count_functions.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        thread2 = Thread(target=http_req, args=(1,))
+        thread1.start()
+        thread2.start()
+        thread1.join()
+        thread2.join()
+        '''time returned from both of the HTTP request should be less than
+        1 since both the request should be processed at the
+        same time because FUNCTIONS_WORKER_PROCESS_COUNT is 2'''
+        time_diff_in_seconds = abs((response[0] - response[1]).total_seconds())
+        self.assertTrue(time_diff_in_seconds < 2)
```
Comment (YunchuWang) [inline tests/endtoend/test_worker_proccess_count_functions.py:124]: less than 1

**Review by YunchuWang — APPROVED** [review]

File: tests/endtoend/test_worker_proccess_count_functions.py (hunk 1 above)
Comment (pdthummar) [reply tests/endtoend/test_worker_proccess_count_functions.py:124]: updated typo. Thanks

File: tests/endtoend/test_worker_proccess_count_functions.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# Licensed under the MIT License.
+import os
+from threading import Thread
+from unittest.mock import patch
+from datetime import datetime
+from tests.utils import testutils
+
+
+class TestWorkerProcessCount1(testutils.WebHostTestCase):
+    """Test the Http Trigger with setting up the python worker process count
```
Comment (gavin-aguiar) [inline tests/endtoend/test_worker_proccess_count_functions.py:11]: Process count is set to 1

File: tests/endtoend/test_worker_proccess_count_functions.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import os
+from threading import Thread
+from unittest.mock import patch
+from datetime import datetime
+from tests.utils import testutils
+
+
+class TestWorkerProcessCount1(testutils.WebHostTestCase):
```
Comment (gavin-aguiar) [inline tests/endtoend/test_worker_proccess_count_functions.py:10]: I don't think we need this test cases. The below testcase of setting fwpc to 2 should be enough to test this feature

File: tests/endtoend/test_threadpool_thread__count_functions.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import os
+from threading import Thread
+from unittest.mock import patch
+from datetime import datetime
+from tests.utils import testutils
+
+
+class TestPythonThreadpoolThreadCount1(testutils.WebHostTestCase):
```
Comment (gavin-aguiar) [inline tests/endtoend/test_threadpool_thread__count_functions.py:10]: Same with this. Setting tptc to 2 should be enough. We dont need this test case

File: tests/endtoend/http_functions/http_functions_stein/function_app.py [hunk 5]
```diff
@@ -32,3 +34,11 @@ def default_template(req: func.HttpRequest) -> func.HttpResponse:
             " personalized response.",
             status_code=200
         )
+
+
+@app.route(route="http_func")
```
Comment (vrdmr) [inline tests/endtoend/http_functions/http_functions_stein/function_app.py:39]: Same comment as before from @gavin-aguiar  on auth level

File: tests/endtoend/http_functions/http_functions_stein/function_app.py [hunk 6]
```diff
@@ -1,7 +1,9 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+from datetime import datetime
 import logging
```
Comment (gavin-aguiar) [inline tests/endtoend/http_functions/http_functions_stein/function_app.py:5]: I don't think logging is needed. Can be removed.

File: tests/endtoend/test_worker_proccess_count_functions.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import os
+from threading import Thread
+from unittest.mock import patch
+from datetime import datetime
+from tests.utils import testutils
+
+
+class TestWorkerProcessCount2(testutils.WebHostTestCase):
```
Comment (gavin-aguiar) [inline tests/endtoend/test_worker_proccess_count_functions.py:10]: nit: You can remove the 2 from all the testnames.

File: tests/endtoend/test_worker_proccess_count_functions.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    """Test the Http Trigger with setting up the python worker process count
+    to 2. this test will check if both requests should be processed at the
+    same time. this file is more focused on testing the E2E flow scenario for
+    FUNCTIONS_WORKER_PROCESS_COUNT feature.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        os_environ = os.environ.copy()
+        os_environ['PYTHON_THREADPOOL_THREAD_COUNT'] = '1'
```
Comment (gavin-aguiar) [inline tests/endtoend/test_worker_proccess_count_functions.py:20]: Do we need this?

**Review by gavin-aguiar — APPROVED** [review]

File: tests/endtoend/http_functions/http_functions_stein/function_app.py (hunk 6 above)
Comment (pdthummar) [reply tests/endtoend/http_functions/http_functions_stein/function_app.py:5]: we are using logging in default_template

Comment (gavin-aguiar) [reply tests/endtoend/http_functions/http_functions_stein/function_app.py:5]: Oh sorry,  didn't see the default_template function

**Pull request merged**